Deriving requires traits in scope

Currently if one does not have the right traits in scope deriving fails:

```rust
error[E0599]: no method named `finish` found for struct `Attributes` in the current scope
   --> src/main.rs:1:10
    |
1   | #[derive(deser::Serialize, deser::Deserialize)]
    |          ^^^^^^^^^^^^^^^^ method not found in `Attributes`
```

--- a/deser-derive/src/ser.rs
+++ b/deser-derive/src/ser.rs
@@ -103,7 +103,7 @@ fn derive_struct(input: &syn::DeriveInput, fields: &syn::FieldsNamed) -> syn::Re
                     #index => {
                         #field_skip
                         if self.nested_emitter_exhausted {
-                            self.nested_emitter = match self.data.#name.serialize(__state)? {
+                            self.nested_emitter = match ::deser::ser::Serialize::serialize(&self.data.#name, __state)? {
                                 ::deser::ser::Chunk::Struct(__inner) => {
                                     Some(__inner)
                                 }
@@ -118,7 +118,7 @@ fn derive_struct(input: &syn::DeriveInput, fields: &syn::FieldsNamed) -> syn::Re
                             ::deser::__derive::None => {
                                 self.index += 1;
                                 self.nested_emitter_exhausted = true;
-                                self.data.#name.finish(__state)?;
+                                ::deser::ser::Serialize::finish(&self.data.#name, __state)?;
                                 continue;
                             }
                             // we need this transmute here because of limitations in the borrow

--- a/deser/tests/test_derive_unscoped.rs
+++ b/deser/tests/test_derive_unscoped.rs
@@ -1,0 +1,22 @@
+#[test]
+fn test_unscoped() {
+    #[derive(deser::Serialize, deser::Deserialize)]
+    #[deser(skip_serializing_optionals)]
+    pub struct Root {
+        flag: Option<bool>,
+        #[deser(flatten)]
+        attrs: Attrs,
+    }
+
+    #[derive(deser::Serialize, deser::Deserialize)]
+    #[deser(skip_serializing_optionals)]
+    pub struct Attrs {
+        is_active: bool,
+        #[deser(skip_serializing_if = "is_false")]
+        is_stuff: bool,
+    }
+
+    fn is_false(value: &bool) -> bool {
+        !*value
+    }
+}

